packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * <TextInput
+  label="Enter Name"
+  labelPosition="left"
+  placeholder="Enter your name"
+  icon={UserIcon}
+  showClearButton
+  helpText="Enter first name and last name. Eg: Kamlesh Chandnani"
+  errorText="Name cannot be blank"
+  successText="Name validated"
+  validationState={inputValue.length < 10 ? 'error' : 'success'}
+/>
+ */
+
+import type { ReactElement } from 'react';
+import { StyledBaseInput } from './StyledBaseInput';
+
+type HandleOnChange = ({
+  event,
+  inputName,
+  value,
+}: {
+  event?: React.ChangeEvent<HTMLInputElement>;
+  inputName?: string;
+  value?: string;
+}) => void;
+
+type OnChange = ({
+  event,
+  name,
+  value,
+}: {
+  event?: React.ChangeEvent<HTMLInputElement>;
+  name?: string;
+  value?: string;
+}) => void;
+
+type LabelProps = {
+  label: string;
+  /**
+   * Desktop only prop. on Mobile by default the label will be on top
+   */
+  labelPosition?: 'top' | 'left';
+};
+
+export type BaseInputProps = LabelProps & {
```

Comment:
There are still some missing props, right? Will be added later?

Reply:
yes, refer milestone chart